client/history/peerResolver.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package history
+
+import (
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/membership"
+	"github.com/uber/cadence/common/service"
+)
+
+// PeerResolver is used to resolve history peers.
+// Those are deployed instances of Cadence history services that participate in the cluster ring.
+// The resulting peer is simply an address of form ip:port where RPC calls can be routed to.
+type PeerResolver struct {
+	numberOfShards int
+	membership     membership.Monitor
+	addressMapper  AddressMapperFn
+}
+
+type AddressMapperFn func(string) (string, error)
+
+// NewPeerResolver creates a new history peer resolver.
+func NewPeerResolver(numberOfShards int, membership membership.Monitor, addressMapper AddressMapperFn) PeerResolver {
+	return PeerResolver{
+		numberOfShards: numberOfShards,
+		membership:     membership,
```

Comment:
If you need only `history` resolver, you can replace membership with `resolver` and pass history resolver here

Reply:
Agree, updated the PR.